system/lib/wasmfs/file_table.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+// This file defines the open file table of the new file system.
+// Current Status: Work in Progress.
+// See https://github.com/emscripten-core/emscripten/issues/15041.
+
+#pragma once
+
+#include "file.h"
+#include "lockable.h"
+#include <assert.h>
+#include <mutex>
+#include <utility>
+#include <vector>
+#include <wasi/api.h>
+
+class OpenFileDescriptor {
```

Comment:
I think this renaming would be good, too.

Reply:
Good point that we should avoid an ambiguity with a numerical index which is a file descriptor. So, such an index is also an "open file descriptor", isn't it? It has to be of an open file. So I don't think "open" is the right way to differentiate these two.

How about calling this "TableEntry" or such? The concept here is really "the table entry's information for an open file. And maybe this can actually be merged with the existing `Entry`?

Reply:
I'm inclined to not worry about naming collisions with user-level file descriptors since they are always identified with their own type `__wasi_fd_t`. I don't think there's much risk of confusion. I think that `TableEntry` risks confusion with `FileTable::Entry`, but I like `FileDescriptor`. This is also consistent with how the documentation describes these concepts: "A call to open() creates a new _open file description_, an entry in the system-wide table of open files."

Reply:
Agreed, FileDescriptor sgtm.

Reply:
I think it makes sense to name it `FileDescription`. I think Sam's comment make sense. I think something other than FileDescriptor would work.